src/Coordination/KeeperStateManager.cpp

```diff
@@ -38,13 +73,24 @@ KeeperStateManager::KeeperConfigurationWrapper KeeperStateManager::parseServersC
         int32_t priority = config.getInt(full_prefix + ".priority", 1);
         bool start_as_follower = config.getBool(full_prefix + ".start_as_follower", false);
 
+        if (client_ports.contains(port))
+        {
+            throw Exception(ErrorCodes::RAFT_ERROR, "Raft config contains hostname '{}' with port '{}' which is equal to client port on current machine",
+                            hostname, port);
+        }
+
+        if (isLocalhost(hostname))
+            local_hostname = hostname;
+        else
+            non_local_hostname = hostname;
```

Comment:
Seems like it will not help if host have multiple network interfaces

Reply:
Yes, but what to do?

Reply:
Seems like `NetworkInterfaces` (`getifaddrs`) allows to get addresses of all interfaces